### DISCUSSION TITLE
WebRTC adaptive bitrate support

### DISCUSSION THREAD

**Discussion by romainreignier** [blakeblackshear/frigate#7782] [Ideas]
First, I would like to thank the authors of this nice piece of software that uses modern technologies, is open source and easy to install thanks to Docker.

From what I understand, the WebRTC view with [go2rtc](https://github.com/AlexxIT/go2rtc) do not support adaptive bitrate. Is that true?
I know that the WebRTC part is handled by [go2rtc](https://github.com/AlexxIT/go2rtc), itself built upon [pion/webrtc](https://github.com/pion/webrtc) so the issue should be raised over there.
But in the mean time, [gstreamer as a complete webrtc suite](https://gstreamer.freedesktop.org/documentation/rswebrtc/index.html?gi-language=c) with adaptive bitrate implemented.

So I was wondering how difficult it would be to have a gstreamer backend for the Webrtc view?

**Comment by bdoiron74** [comment]
I'd like to see support for adaptive bitrate over WebRTC as well. Most cameras support two streams with different encodings, so would it be possible to advertise both and let the WebRTC clients choose the one that works for them based on bandwidth? For example, my camera is configured with a 1920x1080 4Mbps VBR RTSP stream on port 553 and a 960x540 2Mbps VBR stream on port 554.

**Reply by bdoiron74** [reply]
I may have the nomenclature incorrect - I might be proposing '[simulcasting](https://github.com/AlexxIT/go2rtc/issues/288)'
